github/tests/Framework.py

```diff
@@ -89,26 +89,30 @@ def __init__(self, file, protocol, host, port, *args, **kwds):
     def request(self, verb, url, input, headers):
         print verb, url, input, headers,
         self.__cnx.request(verb, url, input, headers)
-        fixAuthorizationHeader(headers)
+        copied_headers = headers.copy() # fixAuthorizationHeader fixes *in place* which breaks requests
```

Comment:
Could you elaborate a bit more on the comments here to help people understand, maybe link the issue number?